core/basic_authorship/block_builder_factory.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_BASIC_AUTHORSHIP_BLOCK_BUILDER_FACTORY_HPP
+#define KAGOME_CORE_BASIC_AUTHORSHIP_BLOCK_BUILDER_FACTORY_HPP
+
+#include "basic_authorship/block_builder.hpp"
+#include "primitives/block_id.hpp"
+#include "runtime/block_builder_api.hpp"
+#include "runtime/core.hpp"
+
+namespace kagome::basic_authorship {
+
+  /**
+   * Creates new block builders
```

Comment:
Maybe add here why BlockBuilder needs a factory and cannot be just created? To make the comment more valuable.